gm4_blast_furnaces/data/blast_furnaces/functions/verify_furnace.mcfunction

```diff
@@ -0,0 +1,11 @@
+# check furnace and store cooktime
+# ran by processes
+
+scoreboard players set @s gm4_bf_data -1
+execute if entity @s[y_rotation=0] if block ~ ~ ~1 furnace[facing=south] store result score @s gm4_bf_data run data get block ~ ~ ~1 CookTime
+execute if entity @s[y_rotation=90] if block ~-1 ~ ~ furnace[facing=west] store result score @s gm4_bf_data run data get block ~-1 ~ ~ CookTime
+execute if entity @s[y_rotation=180] if block ~ ~ ~-1 furnace[facing=north] store result score @s gm4_bf_data run data get block ~ ~ ~-1 CookTime
+execute if entity @s[y_rotation=-90] if block ~1 ~ ~ furnace[facing=east] store result score @s gm4_bf_data run data get block ~1 ~ ~ CookTime
+
+# display ore
+execute if score @s gm4_bf_data matches 0.. run function blast_furnaces:check_ore
```

Comment:
Move this call to `processes` so you don't need this function and you don't need to use the `gm4_bf_data` scoreboard to check if it has a furnace.